packages/core/src/request.ts

```diff
@@ -159,8 +164,50 @@ export class Request {
         this.retryCount = retryCount;
         this.errorMessages = [...errorMessages];
         this.headers = { ...headers };
-        this.userData = { ...userData };
         this.handledAt = handledAt as unknown instanceof Date ? (handledAt as Date).toISOString() : handledAt!;
+
+        Object.defineProperties(this, {
+            _userData: {
+                value: { ...userData },
```

Comment:
If the `userData` object comes from API with `__crawlee` defined, we will add it to `this._userData`, but won't set the property non-enumerable, so if the user then retrieves it with the `userData` getter, they will see it.

Reply:
Good catch. We can either redefine the property to be non enumerable, or use the setter as you suggested.

```ts
const o = { foo: 123 };
console.log(Object.keys(o));
Object.defineProperty(o, 'foo', { enumerable: false, writable: true });
console.log(Object.keys(o));
```